source/casts/utils/index.ts

```diff
@@ -40,6 +40,33 @@ export const indexToMapCoordinate = (index, resolution = 1): IGeoLocation => {
 	const latitude = -90 + (location.y - 0.5) / resolution;
 	return { longitude, latitude };
 };
+
+// Handle date strings on the format YYYYMMDD
+export const convertStringToDate = (dateString: string): Date => {
+	let date;
+
+	if (dateString.length < 4) {
+		throw new Error("Unknown date format");
+	}
+	try {
+		if (dateString.length === 4) {
+			// only year
+			date = new Date(dateString);
+		} else if (dateString.length < 8) {
+			date = new Date(parseInt(dateString.slice(0, 4), 10), parseInt(dateString.slice(5, 6), 10) - 1);
```

Comment:
This is starting to look a bit repetitive, how about something like:
```
const dateComponents: number[] = []

// parse year
if (dateString.length >= 4) {
  dateComponents.push(parseInt(dateString.slice(0, 4), 10))
}

// parse month
if (dateString.length >= 6) {
  dateComponents.push(parseInt(dateString.slice(5, 6), 10) - 1))
}

// parse day
if (dateString.length >= 8 {
  dateComponents.push(parseInt(dateString.slice(7, 8), 10))
}

// ensure date components were parsed correctly
if (dateComponents.filter(isNaN).length > 0) {
  throw new Error(`Failed to parse "${dateString}" into a valid date`)
}

// create new date and validate
const date = new Date(...dateComponents)
if (isNaN(date)) {
  throw new Error(`Failed to parse "${dateString}" into a valid date`)
}

return date
```

`parseInt` and the `Date` constructor do not throw errors, so we need to validate the return values ourselves.